File: src/C++/task_2/pangram.cpp
#include <iostream>
#include <string>
#include <cctype> // для функции tolower
using namespace std;

bool is_pangram(const string& sentence) {
    bool letters[26] = {false}; // Массив для отслеживания встреченных букв

    for (char ch : sentence) {
        if (isalpha(ch)) { // Проверяем, является ли символ буквой
            letters[tolower(ch) - 'a'] = true; // Отмечаем в массиве
        }
    }

    // Проверяем, встретились ли все буквы
    for (bool exists : letters) {
        if (!exists) return false; // Если найдена буква, которая не встретилась, возвращаем false
    }

    return true; // Все буквы встретились, значит, это панграмма
}

int main() {
    string sentence;
    getline(cin, sentence); // Считываем предложение со стандартного ввода

    cout << (is_pangram(sentence) ? "True" : "False") << endl; // Выводим результат

    return 0;
}

File: src/JavaScript/task_2/pangram.js
function isPangram(sentence) {
    // Убираем пробелы и приводим строку к нижнему регистру
    let formattedSentence = sentence.replace(/\s+/g, '').toLowerCase();
    
    // Создаем набор уникальных символов
    const letters = new Set(formattedSentence);
    
    // Проверяем, что количество уникальных букв не меньше 26 (количество букв в английском алфавите)
    return letters.size >= 26;
  }
  
  // Ввод строки пользователем через всплывающее окно
  let sentence = prompt();
  console.log(isPangram(sentence));
  